### PR TITLE
[config] move config.ActPool to actpool package refactor

### LINKED ISSUE #3513: move config.ActPool to actpool package refactor
as title

### Why is this needed
- [] better/less memory usage
- [] reduce running time
- [] clean (easy-to-understand) data structure or algorithm
- [] remove dead code
- [x] better code design or structure (like reduce complexity)
- [] reduce/decouple dependencies between code modules
- [] increase test coverage
- [x] better code readability

### PR DESCRIPTION
**Pull request by millken**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #3513

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/protocol/rewarding/reward_test.go [hunk 1]
```diff
@@ -382,18 +383,18 @@ func TestProtocol_NoRewardAddr(t *testing.T) {
...
+		g.UnproductiveDelegateMaxCacheSize,
+		g.ProbationIntensityRate)
 	require.NoError(t, err)
 	pp, err := poll.NewGovernanceChainCommitteeProtocol(
 		nil,
 		committee,
 		uint64(123456),
 		func(uint64) (time.Time, error) { return time.Now(), nil },
-		cfg.Chain.PollInitialCandidatesInterval,
+		bcf.PollInitialCandidatesInterval,
```
Comment (dustinxie) [inline action/protocol/rewarding/reward_test.go:397]: directly use `blockchain.DefaultConfig.PollInitialCandidatesInterval` here, no need to define `bcf`

File: config/config_test.go [hunk 2]
```diff
@@ -312,7 +313,7 @@ func TestValidateActPool(t *testing.T) {
 }
 
 func TestValidateMinGasPrice(t *testing.T) {
-	ap := ActPool{MinGasPriceStr: Default.ActPool.MinGasPriceStr}
+	ap := actpool.Config{MinGasPriceStr: Default.ActPool.MinGasPriceStr}
```
Comment (dustinxie) [inline config/config_test.go:316]: move this test to `actpool` package

File: action/protocol/poll/protocol.go [hunk 3]
```diff
@@ -19,13 +19,16 @@ import (
...
+	"github.com/iotexproject/iotex-core/blockchain"
+	"github.com/iotexproject/iotex-core/blockchain/genesis"
 	"github.com/iotexproject/iotex-core/pkg/log"
 	"github.com/iotexproject/iotex-core/state"
 )
 
 const (
 	_protocolID = "poll"
+	// RollDPoSScheme means randomized delegated proof of stake
+	RollDPoSScheme = "ROLLDPOS"
```
Comment (dustinxie) [inline action/protocol/poll/protocol.go:31]: `_rollDPoSScheme` no need to export

**Comment by Liuhaai** [comment]
The changes on `poll` protocol is unrelated to this pr. Pls remove them from here. Other changes looks good for me

**Comment by millken** [comment]
> The changes on `poll` protocol is unrelated to this pr. Pls remove them from here. Other changes looks good for me

If no change here,  will get an "import cycle not allowed" error on build

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**